src/runtime/formal/ops/ops.cc

```diff
@@ -249,9 +249,12 @@ CVM_REGISTER_GLOBAL("cvm.runtime.formal.max_pool2d")
       for (int s = 0; s < filter_w; ++s) {
         int32_t tp = p * stride_h + r - padding[0];
         int32_t tq = q * stride_w + s - padding[1];
-        int32_t x_tmp = minV; 
-        if (0 <= tp && tp < x_h && 0 <= tq && tq < x_w)
-          x_tmp = GETX(n, k, tp, tq);
+        int32_t x_tmp = minV;
+        if (0 <= tp && tp < x_h && 0 <= tq && tq < x_w) {
+          Indices xIdx(xShape, {n, k, tp, tq});
```

Comment:
It's no need to pass the index of `n, k, tp, tq` to construct the `xIdx`, just use the
`CopyFromIndices` function please. And remember to pass the `yIdx` reference
to the function instead of the index value.